SuperBuild/External_aruco.cmake

```diff
@@ -0,0 +1,79 @@
+	
+    IF(MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode")
+      IF(EXISTS ${ITK_RELEASE_FILE})
+        FILE(COPY ${ITK_RELEASE_FILE} DESTINATION ${PLUS_EXECUTABLE_OUTPUT_PATH}/Release)
+      ENDIF()
+      IF(EXISTS ${ITK_DEBUG_FILE})
+        FILE(COPY ${ITK_DEBUG_FILE} DESTINATION ${PLUS_EXECUTABLE_OUTPUT_PATH}/Debug)
+      ENDIF()
+    ELSE()
+      IF(ITK_DEBUG_FILE EQUAL ITK_RELEASE_FILE AND EXISTS ${ITK_RELEASE_FILE})
+        FILE(COPY ${ITK_RELEASE_FILE} DESTINATION ${PLUS_EXECUTABLE_OUTPUT_PATH})
+      ELSE()
+        IF(EXISTS ${ITK_RELEASE_FILE})
+          FILE(COPY ${ITK_RELEASE_FILE} DESTINATION ${PLUS_EXECUTABLE_OUTPUT_PATH})
+        ENDIF()
+        IF(EXISTS ${ITK_DEBUG_FILE})
+          FILE(COPY ${ITK_DEBUG_FILE} DESTINATION ${PLUS_EXECUTABLE_OUTPUT_PATH})
+        ENDIF()
+      ENDIF()
+    ENDIF()
+  ENDFOREACH()
+  
+  SET (PLUS_aruco_DIR ${aruco_DIR} CACHE INTERNAL "Path to store aruco binaries")
+  
+ELSE()
+  #SET(OpenCV_DIR ${PLUS_OpenCV_DIR})
+  # aruco has not been built yet, so download and build it as an external project
+  SET(aruco_GIT_REPOSITORY https://github.com/PlusToolkit/aruco.git)
+  
+  MESSAGE(STATUS "Downloading aruco from: ${aruco_GIT_REPOSITORY}")
+  
+  SET (PLUS_aruco_src_DIR ${CMAKE_BINARY_DIR}/Deps/aruco CACHE INTERNAL "Path to store aruco contents")
+  #SET (PLUS_aruco_LIBS_DIR ${CMAKE_BINARY_DIR}/Deps/aruco CACHE INTERNAL "Path to store aruco contents")
+  SET (PLUS_aruco_prefix_DIR ${CMAKE_BINARY_DIR}/Deps/aruco-prefix CACHE INTERNAL "Path to store aruco prefix data.")
+  SET (PLUS_aruco_DIR ${CMAKE_BINARY_DIR}/Deps/aruco-bin CACHE INTERNAL "Path to store aruco binaries.")
+  SET (PLUS_aruco_install_DIR $CMAKE_BINARY_DIR/Deps/aruco-install CACHE INTERNAL "Path to store aruco install.")
+  
+  ExternalProject_Add( aruco
+    PREFIX ${PLUS_aruco_prefix_DIR}
+    "${PLUSBUILD_EXTERNAL_PROJECT_CUSTOM_COMMANDS}"
+    SOURCE_DIR "${PLUS_aruco_src_DIR}"
+    BINARY_DIR "${PLUS_aruco_DIR}"
+    #--Download step--------------
+    GIT_REPOSITORY ${aruco_GIT_REPOSITORY}
+    #--Configure step-------------
+    CMAKE_ARGS
+      ${ep_common_args}
+      ${aruco_PLATFORM_SPECIFIC_ARGS}
+      -DCMAKE_RUNTIME_OUTPUT_DIRECTORY:PATH=${PLUS_EXECUTABLE_OUTPUT_PATH}
+      -DCMAKE_LIBRARY_OUTPUT_DIRECTORY:PATH=${PLUS_LIBRARY_OUTPUT_PATH}
+      -DCMAKE_ARCHIVE_OUTPUT_DIRECTORY:PATH=${PLUS_ARCHIVE_OUTPUT_PATH}
+      -DOpenCV_INSTALL_BINARIES_PREFIX:STRING= # Install to prefix directly, not arch/compiler/etc...
+      -DCMAKE_CXX_FLAGS:STRING=${ep_common_cxx_flags}
+      -DCMAKE_C_FLAGS:STRING=${ep_common_c_flags}
+      -DOpenCV_DIR:PATH=${PLUS_OpenCV_DIR}
+      -DUSE_OWN_EIGEN3=OFF
+      -DBUILD_TESTS:BOOL=OFF
+      -DBUILD_PERF_TESTS:BOOL=OFF
+      -DBUILD_SHARED_LIBS:BOOL=${PLUSBUILD_BUILD_SHARED_LIBS}
+    #--Build step-----------------
+    BUILD_ALWAYS 1
+    #--Install step-----------------
+    INSTALL_COMMAND "" # don't install
+    #--Dependencies-----------------
+    DEPENDS OpenCV
```

Comment:
This will probably break if the user specifies an OpenCV_DIR as it will have targets such as opencv_core (I forget their exact naming).

Your two options are to create a dummy target in External_Opencv.cmake named OpenCV which does nothing, or to have some logic above here targetting a valid OpenCV target.